File: token-client-spring/src/main/java/no/nav/security/token/support/oauth2/client/OAuth2AccessTokenResponse.java
package no.nav.security.token.support.oauth2.client;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import no.nav.security.token.support.oauth2.OAuth2ParameterNames;

import java.util.HashMap;
import java.util.Map;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class OAuth2AccessTokenResponse {
    @JsonProperty(OAuth2ParameterNames.ACCESS_TOKEN)
    private String accessToken;
    @JsonProperty(OAuth2ParameterNames.EXPIRES_AT)
    private int expiresAt;
    @JsonAlias({"expires_in", "ext_expires_in"})
    private int expiresIn;
    @JsonAnySetter
    private Map<String, Object> additonalParameters = new HashMap<>();
}

File: token-client-core/src/main/kotlin/no/nav/security/token/support/client/core/oauth2/OAuth2AccessTokenResponse.kt
package no.nav.security.token.support.client.core.oauth2

data class OAuth2AccessTokenResponse (@get:JvmName("getAccessToken") var access_token : String? = null,
                                      @get:JvmName("getExpiresAt") var expires_at : Int? = null,
                                      @get:JvmName("getExpiresIn") var expires_in : Int? = 60,
                                      private val additionalParameters : Map<String, Any> = emptyMap()) {


    fun getAccess_token() = access_token
    fun getExpires_at() = expires_at
    fun getExpires_in() = expires_in
}